pkg/log/logger.go

```diff
@@ -86,3 +88,39 @@ func newZapLoggerTo(destWriter io.Writer, level LogLevel, opts ...zap.Option) *z
 	return zap.New(zapcore.NewCore(&kube_log_zap.KubeAwareEncoder{Encoder: enc, Verbose: level == DebugLevel}, sink, lvl)).
 		WithOptions(opts...)
 }
+
+type loggerCtx struct{}
+
+// WithFields will add to provided context fields (key/value pairs) which then
+// will be logged by logger decorated with DecorateWithCtx function.
+// After adding logging fields it returns back enriched context.
+func WithFields(ctx context.Context, keysAndValues ...interface{}) context.Context {
+	fields, ok := ctx.Value(loggerCtx{}).(*[]interface{})
+	if !ok || fields == nil {
+		return context.WithValue(ctx, loggerCtx{}, &keysAndValues)
+	}
+
+	*fields = append(*fields, keysAndValues...)
+
+	return ctx
+}
+
+// DecorateWithCtx will check if provided context contain tracing span and
+// if the span is currently recording. If so, it will add trace_id and span_id
+// to logged values. It will also add to logger values from fields added to
+// context earlier by WithFields functions.
+func DecorateWithCtx(logger logr.Logger, ctx context.Context) logr.Logger {
+	if span := trace.SpanFromContext(ctx); span.IsRecording() {
+		logger = logger.WithValues(
+			"trace_id", span.SpanContext().TraceID(),
+			"span_id", span.SpanContext().SpanID(),
```

Comment:
nit but I'll ask anyways. Our convention for keys in log is camelCase. Is there a reason you use snake_case here?

Reply:
Yes, that's the property name used to extract trace ids by datadog, dynatrace etc.